File: app/src/main/java/com/doapps/habits/listeners/NameChangeListener.java
package com.doapps.habits.listeners;

import java.util.Observable;


public class NameChangeListener extends Observable {
    private boolean mChanged;
    public static final NameChangeListener listener = new NameChangeListener();

    public void setChanged() {
        mChanged = true;
        super.notifyObservers();
    }

    @Override
    public boolean hasChanged() {
        synchronized (this) {
            return mChanged;
        }
    }
}

File: app/src/main/java/com/doapps/habits/listeners/NameChangeListener.kt
package com.doapps.habits.listeners

import java.util.*


class NameChangeListener : Observable() {
  private var mChanged = false

  public override fun setChanged() {
    mChanged = true
    super.notifyObservers()
  }

  override fun hasChanged(): Boolean {
    synchronized(this) {
      return mChanged
    }
  }

  companion object {
    val listener = NameChangeListener()
  }
}
